playground/pspin/start_connect_worker.py

```diff
@@ -0,0 +1,36 @@
+import argparse
+from pele.concurrent import ConnectWorker
+from start_server import create_system, get_server_uri
+from pele.storage import Database
+
+def main():
+    parser = argparse.ArgumentParser(description="connect worker queue")
+    parser.add_argument("p", type=int, help="p-spin")
+    parser.add_argument("nspins", type=int, help="number of spins")
+    parser.add_argument("--strategy", type=str, help="strategy to adopt: random (default), "
+                                                     "untrap, combine, gmin", default="random")
+    args = parser.parse_args()
+
+    nspins = args.nspins
+    p = args.p
+
+    #get interactions from database
+    dbname = "pspin_spherical_p{}_N{}.sqlite".format(p,nspins)
+    db = Database(dbname, createdb=False)
+    interactions = db.get_property("interactions").value()
+    db_nspins = db.get_property("nspins").value()
+    db_p = db.get_property("p").value()
+    assert db_nspins == nspins
+    assert db_p == p
+    #close this SQLAlchemy session
+    db.session.close()
```

Comment:
good point!

On Thu, Jan 7, 2016 at 8:20 AM, Jacob Stevenson <notifications@github.com>
wrote:

> In playground/pspin/start_connect_worker.py
> <https://github.com/pele-python/pele/pull/140#discussion_r49049963>:
>
> > +                                                     "untrap, combine, gmin", default="random")
> > +    args = parser.parse_args()
> > +
> > +    nspins = args.nspins
> > +    p = args.p
> > +
> > +    #get interactions from database
> > +    dbname = "pspin_spherical_p{}_N{}.sqlite".format(p,nspins)
> > +    db = Database(dbname, createdb=False)
> > +    interactions = db.get_property("interactions").value()
> > +    db_nspins = db.get_property("nspins").value()
> > +    db_p = db.get_property("p").value()
> > +    assert db_nspins == nspins
> > +    assert db_p == p
> > +    #close this SQLAlchemy session
> > +    db.session.close()
>
> It's good that you close the db to avoid running into problems having
> multiple instances open. To be even more safe you might want to put this
> section in a separate function so db can be garbage collected.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pele-python/pele/pull/140/files#r49049963>.
>